ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Settings.hs

```diff
@@ -0,0 +1,270 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.Settings
+    ( Settings
+    , initSettings
+    , showSettingsWindow
+    ) where
+
+import Control.Lens (makeLensesWith)
+import Data.Bits
+import Data.Version (showVersion)
+import Web.Browser (openBrowser)
+
+import Graphics.UI.Qtah.Core.Types (alignHCenter, alignVCenter)
+import Graphics.UI.Qtah.Signal (connect_)
+import Graphics.UI.Qtah.Widgets.QSizePolicy (QSizePolicyPolicy(..))
+
+import qualified Graphics.UI.Qtah.Core.QObject as QObject
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QComboBox as QComboBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLayout as QLayout
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QVBoxLayout as QVBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+import Ariadne.Version (currentAriadneVersion)
+
+data Settings =
+  Settings
+    { settings :: QDialog.QDialog
+    , generalButton :: QPushButton.QPushButton
+    , aboutButton :: QPushButton.QPushButton
+    , supportButton :: QPushButton.QPushButton
+    , generalWidget :: QWidget.QWidget
+    , generalSettings :: GeneralSettings
+    , aboutWidget :: QWidget.QWidget
+    , aboutSettings :: AboutSettings
+    , supportWidget :: QWidget.QWidget
+    , supportSettings :: SupportSettings
+    }
+
+data SettingsMode = GeneralMode | AboutMode | SupportMode deriving (Show, Eq, Enum, Bounded)
+
+data GeneralSettings =
+  GeneralSettings
+    { languageSelector :: QComboBox.QComboBox
+    , countervalueSelector :: QComboBox.QComboBox
+    , rateProviderSelector :: QComboBox.QComboBox
+    , themeSelector :: QComboBox.QComboBox
+    }
+
+data AboutSettings =
+  AboutSettings
+    { releaseNotesButton :: QPushButton.QPushButton
+    , licenseButton :: QPushButton.QPushButton
+    }
+
+data SupportSettings =
+  SupportSettings
+    { faqButton :: QPushButton.QPushButton
+    , clearCacheButton :: QPushButton.QPushButton
+    , resetButton :: QPushButton.QPushButton
+    }
+
+makeLensesWith postfixLFields ''Settings
+
+initSettings :: IO (QDialog.QDialog, Settings)
+initSettings = do
+  settings <- QDialog.new
+  QWidget.resizeRaw settings 776 363
+
+  QObject.setObjectName settings ("settingsDialog" :: String)
+  QWidget.setWindowTitle settings ("Settings" :: String)
+  QWidget.adjustSize settings
+
+  settingsLayout <- createLayout settings
+  QLayout.setContentsMarginsRaw settingsLayout 0 24 0 0
+  QLayout.setSpacing settingsLayout 0
```

Comment:
Why are these two lines needed?

Reply:
first line is provide indentation due to zeplin

Reply:
Well, `createLayout` does that already. Am I getting confused again?

Reply:
first is not really needed, second reduce distance between header and row with buttons

Reply:
Okay, fair enough. This reminds me of another problem -- please set spacing to 18 in the actual widgets' layouts, so that rows are not squished one after another.